lang/cs/Org.Apache.REEF.IMRU/InProcess/InProcessIMRUClient.cs

```diff
@@ -52,9 +54,32 @@ private InProcessIMRUClient([Parameter(typeof(NumberOfMappers))] int numberOfMap
             _numberOfMappers = numberOfMappers;
         }
 
+        /// <summary>
+        /// Submits the map job
+        /// </summary>
+        /// <param name="jobDefinition">Job definition given by the user</param>
+        /// <returns>The result of the job</returns>
         public IEnumerable<TResult> Submit(IMRUJobDefinition jobDefinition)
         {
-            var injector = TangFactory.GetTang().NewInjector(jobDefinition.Configuration);
+            var emptyConfig = TangFactory.GetTang().NewConfigurationBuilder().Build();
+            var updateFunctionCodecConfig = jobDefinition.UpdateFunctionCodecsConfiguration;
+            var mapInputPipelineConfig = jobDefinition.MapInputPipelineDataConverterConfiguration;
+            var mapOutputPipelineConfig = jobDefinition.MapOutputPipelineDataConverterConfiguration;
+
+            if (typeof (TMapInput) == typeof (TMapOutput))
+            {
+                mapOutputPipelineConfig = emptyConfig;
+            }
+
+            var mergedConfig = Configurations.Merge(
```

Comment:
Given the reasoning above, we should consider keeping them separate even in the local execution model. Maybe as a separate JIRA?